sycl/source/detail/queue_impl.hpp

```diff
@@ -439,27 +439,41 @@ class queue_impl {
     return MAssertHappenedBuffer;
   }
 
-private:
-  void finalizeHandler(handler &Handler, const CG::CGTYPE &Type,
+protected:
+  // template is needed for proper unit testing
+  template <typename HandlerType = handler>
+  void finalizeHandler(HandlerType &Handler, const CG::CGTYPE &Type,
                        event &EventRet) {
     if (MIsInorder) {
-      bool NeedSeparateDependencyMgmt =
-          (Type == CG::CGTYPE::CodeplayHostTask ||
-           Type == CG::CGTYPE::CodeplayInteropTask);
+
+      auto IsExpDepManaged = [](const CG::CGTYPE &Type) {
```

Comment:
Just curious, what `Exp` stands for here?

Reply:
explicitly)